flutter_appauth/ios/Classes/FlutterAppauthPlugin.m

```diff
@@ -483,12 +483,16 @@ - (BOOL)application:(UIApplication *)application
             openURL:(NSURL *)url
             options:
                 (NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options {
-  if ([_currentAuthorizationFlow resumeExternalUserAgentFlowWithURL:url]) {
-    _currentAuthorizationFlow = nil;
-    return YES;
+  @try {
```

Comment:
I'm having trouble getting the timing right to reproduce this but based on the code changes, from the perspective of an app using the plugin from the Dart/Flutter side, the exception is silently caught and won't be informed that there was a problem. Is my understanding correct? If it is, can the PR make the appropriate changes that would raise an exception for apps to handle? The other thing is based on what you mentioned, I suspect macOS would also have a similar problem

Reply:
This bug is somewhat rare, occurring about 1 to 2 times out of 20. I'd prefer to ignore this exception if possible, but the app crashes if it's not handled in objective-c. It works fine even if we don't raise the exception to Dart/Flutter, as long as it's managed in Objective-C. For macOS, I believe it would face the same issue if it uses the same login mechanism.

Reply:
I added a simple throw in the catch functions and also added try catch to macOS as well. 
Please let me know if you have any suggestion or question.

Reply:
> Edit: Do you have a suggestion for an exception type to raise?

The plugin will handle throwing the appropriate exception provided you make the calls on the native method channel side. There are existing examples do that this on iOS and macOS by calling [this method](https://github.com/MaikuB/flutter_appauth/blob/1ceea0e7a322405d3749395f4ed4aa8620330daa/flutter_appauth/ios/Classes/FlutterAppAuth.m#L53). Calling this will raise the appropriate exception

I understand this is rare and how the PR in its current form avoids a crashed but since it's silently caught from the perspective of the Flutter side, it doesn't provide the app/developer the opportunity to catch the exception and leave it them to do decide what to do. If I'm not mistaken, the PR as it is would prevent an uncaught exception from going to something like Crashlytics too

Reply:
As you mentioned, the exception is typically handled in the code you provided. However, this specific error originates from the code [here](https://github.com/MaikuB/flutter_appauth/blob/1ceea0e7a322405d3749395f4ed4aa8620330daa/flutter_appauth/ios/Classes/FlutterAppauthPlugin.m#L481C1-L492C2). I've attached a video that should clarify things. In the video, I simply click the "done" button during the auto login process (no password entry required). There's a slight chance the app might either crash or freeze (in the video, it froze), and you can see the exception printed on the left side of the console.

https://github.com/user-attachments/assets/8366323b-4734-478e-8078-72a8fbaa6893

You're right in your understanding, but I'm uncertain about how can I help debug when this happens. My code is intended to prevent the app from crashing only. Could you either demonstrate or guide me on the proper way to handle this?

Reply:
To clarify, I wasn't asking for helping to debug. I was asking for the code to be changed so that when the scenario happens, rather than silently swallowing the error, it results in an except that can by caught by applications. I understand this is rare but this would enable applications to know something happened and decide what to do. In terms of guidance, refer to the links in my previous post as the intent there to provide references to existing that shows how it's currently done to help guide. It has links to code that shows how the plugin raises exceptions starting from the native side and how that makes its way to the Flutter side

Reply:
OK, I see, thanks for your clarification.